src/main/core/manager.rs

```diff
@@ -684,65 +691,6 @@ impl<'a> Manager<'a> {
         Ok(host)
     }
 
-    // assume that the provided env variables are UTF-8, since working with str instead of OsStr is
-    // much less painful
-    fn generate_env_vars(&self, env: BTreeMap<EnvName, String>) -> Vec<OsString> {
-        let mut env: BTreeMap<EnvName, OsString> =
-            env.into_iter().map(|(k, v)| (k, v.into())).collect();
-
-        // also insert the plugin preload entries
-        // precendence here is:
-        //   - preload path of the injector
-        //   - preload path of the libc lib
-        //   - preload path of the openssl rng lib
-        //   - preload path of the openssl crypto lib
-        //   - preload values from LD_PRELOAD entries in the environment process option
-
-        let preload = self.make_preload_paths();
-
-        for path in &preload {
-            let path = path.as_os_str().as_bytes();
-            // these two characters separate paths and aren't valid in a preload path
-            assert!(!path.contains(&b':'));
-            assert!(!path.contains(&b' '));
-        }
```

Comment:
I think we should keep this check somewhere.

Reply:
Done